lib/compat/wordpress-6.1/script-loader.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Load global styles assets in the front-end.
+ *
+ * @package gutenberg
+ */
+
+function filter_out_block_nodes( $nodes ) {
+	return array_filter( $nodes, function( $node, $key ) {
+		return ! in_array( 'blocks', $node['path'] );
+	}, ARRAY_FILTER_USE_BOTH );
+}
+
+/**
+ * Enqueues the global styles defined via theme.json.
+ * This should replace wp_enqueue_global_styles.
+ *
+ * @since 5.8.0
+ */
+function gutenberg_enqueue_global_styles() {
+	$separate_assets  = wp_should_load_separate_core_block_assets();
+	$is_block_theme   = wp_is_block_theme();
+	$is_classic_theme = ! $is_block_theme;
+
+	/*
+	 * Global styles should be printed in the head when loading all styles combined.
+	 * The footer should only be used to print global styles for classic themes with separate core assets enabled.
+	 *
+	 * See https://core.trac.wordpress.org/ticket/53494.
+	 */
+	if (
+		( $is_block_theme && doing_action( 'wp_footer' ) ) ||
+		( $is_classic_theme && doing_action( 'wp_footer' ) && ! $separate_assets ) ||
+		( $is_classic_theme && doing_action( 'wp_enqueue_scripts' ) && $separate_assets )
+	) {
+		return;
+	}
+
+	if ( $separate_assets ) {
```

Comment:
Aside from this block, this whole function is copypasta.

Reply:
Let's add a comment explaining this code branch.